### PR TITLE
Integrate transactions `/transactions/estimate-fees` endpoint

### LINKED ISSUE #1801: Integrate transactions `/transactions/estimate-fees` endpoint
### Description

Integrate https://github.com/LiskHQ/lisk-service/issues/1573 and replace with existing fee calculation logic once service team is done with adding the endpoint to the API.

### Motivation

Make better and more consistent the transaction fee calculations.

### PR DESCRIPTION
**Pull request by ClementeSerrano**
### What was the problem?

This PR resolves #1801 and #1833

### How was it solved?

- [x] Add `useTransactionEstimateFeesMutation` hook.
- [x] Add `useTransactionFees` hook.
- [x] Integrate new fees estimation hooks with token:transfer form.
- [x] Improve fees breakdown calculation and rendering.
- [x] Remove all old fees estimation hooks and business logic.
- [x] Fix fees breakdown rendering.
- [x] Handle properly fees estimations loading and error states on UI.
- [x] Fix no-balance account can't receive tokens by positive-balance accounts. Fixed issue related to When an account has no balance and we access to its profile from another current account, the send token button is always disabled.
- [x] Fix new transactions events handling for current accounts. Now filtering the event by current account and only performing the queries update when event is related to that account
- [x]  Fix Transactions overview shows more than 3 transactions when a new transaction is submitted.

### How was it tested?

- [x] iOS Simulator.
- [x] Android Emulator.

### CODE REVIEW

File: src/modules/Transactions/hooks/useTransactionFees.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  //   console.log({ areParamsValid, isTransactionSuccess });
+
+  useEffect(() => {
+    if (isTransactionSuccess && areParamsValid) {
+      // eslint-disable-next-line no-unused-vars
+      const { signatures, ...transactionJSON } = transaction.toJSON();
+
+      estimateFeesMutation.mutate({
+        transaction: transactionJSON,
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/Transactions/hooks/useTransactionFees.js:76]: for the `/estimate-fees` endpoint, unfortunately we can't send it like that. We don't have to send the field at all

File: src/modules/SendToken/hooks/useSendTokenForm.js [hunk 2]
```diff
@@ -198,6 +191,8 @@ export default function useSendTokenForm({ transaction, isTransactionSuccess, in
       const defaultTokenID = applicationSupportedTokensData[0]?.tokenID;
 
       if (defaultTokenID) {
+        console.log('reseting token id to: ', defaultTokenID);
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/SendToken/hooks/useSendTokenForm.js]: done

File: src/modules/SendToken/hooks/useSendTokenForm.js [hunk 3]
```diff
@@ -87,21 +83,18 @@ export default function useSendTokenForm({ transaction, isTransactionSuccess, in
...
-    tokenID,
-    enabled: false,
-    isCrossChainTransfer,
+  useTransactionFees({
+    transaction,
+    isTransactionSuccess,
+    dependencies: [recipientAddress, tokenID, amount],
   });
 
+  console.log(transaction);
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/SendToken/hooks/useSendTokenForm.js]: done

File: src/modules/Accounts/components/AccountDetails/AccountDetails.js [hunk 4]
```diff
@@ -37,7 +37,9 @@ export default function AccountDetails({ account }) {
     data: accountCanSendTokensData,
     isLoading: isLoadingAccountCanSendTokens,
     isError: isErrorAccountCanSendTokens,
-  } = useAccountCanSendTokens(account.address);
+  } = useAccountCanSendTokens(
+    !isCurrentAccount ? currentAccount.metadata.address : account.address
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/Accounts/components/AccountDetails/AccountDetails.js:41]: cause account details component can be rendered from 2 parts: on the home screen to show the details of a current logged in account, or to show the account details of another account (not the current logged in one). We need to check that for determining if the send button is enabled or not.

File: src/modules/Transactions/hooks/useTransactionFees.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      }
+
+      if (data.dynamicFeeEstimates) {
+        updates = { ...updates, dynamicFeeEstimates: data.dynamicFeeEstimates };
+      }
+
+      transaction.update(updates);
+    },
+    onError: (error) => {
+      console.log('mutationError', error);
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/Transactions/hooks/useTransactionFees.js]: done

File: src/modules/SendToken/components/SelectTokenStep/components.js [hunk 6]
```diff
@@ -284,36 +277,29 @@ export function SendTokenPriorityField({ value, onChange, style }) {
...
+        {priorities?.map((priority) => (
+          <TouchableOpacity
+            key={priority.code}
+            onPress={() => onChange(priority.code)}
+            style={[
+              styles.priorityButtonBase,
+              styles[
+                value === priority.code ? 'selectedPriorityButton' : 'notSelectedPriorityButton'
+              ],
+              { marginRight: 8 },
```
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/SendToken/components/SelectTokenStep/components.js]: done

File: src/modules/Accounts/components/AccountDetails/AccountDetails.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ClementeSerrano) [reply src/modules/Accounts/components/AccountDetails/AccountDetails.js]: ok done

File: src/modules/SendToken/hooks/useSendTokenForm.js [hunk 7]
```diff
@@ -241,20 +233,15 @@ export default function useSendTokenForm({ transaction, isTransactionSuccess, in
...
       transaction.update({
         command: 'transferCrossChain',
         params: {
-          messageFee,
-          messageFeeTokenID: recipientApplicationChainChannelData.data.messageFeeTokenID,
           receivingChainID: recipientApplicationChainID,
+          // TODO: Waiting https://github.com/LiskHQ/lisk-service/issues/1669 to be solved
+          // so can be handled by useTransactionFees hook.
+          messageFee: '10000000',
+          messageFeeTokenID: '0400000000000000',
```
Comment (ManuGowda) [inline src/modules/SendToken/hooks/useSendTokenForm.js:244]: why is this hardcoded?

Comment (ClementeSerrano) [reply src/modules/SendToken/hooks/useSendTokenForm.js:244]: as we talked last week, we need to hard code that values until service fixes the endpoint to be able to run without specifying them. Right now we need to pass this dummy values to make the endpoint work.

After making the API call, we receive the correct values of message fees from service and update them accordingly on the form and transaction class. But we need first to send dummy values to the api

Comment (ManuGowda) [reply src/modules/SendToken/hooks/useSendTokenForm.js:244]: Ok, then please remember to add this point in the new issue so that we can update it

**Pull request merged**